common/lib/dependabot/shared_helpers.rb

```diff
@@ -333,7 +343,7 @@ def self.run_shell_command(command, allow_unsafe_shell_command: false, env: {},
       }
 
       raise SharedHelpers::HelperSubprocessFailed.new(
-        message: stdout,
+        message: stderr_to_stdout ? stdout : "#{stderr}\n#{stdout}",
```

Comment:
This _should_ be fine, but I think we can end up with some error messages printed out of order? Not sure what a better approach is

Reply:
Yes, that's correct. But I don't think it's a big deal? I don't have a better approach either.

For the particular command where I plan to use this (`swift package show-dependencies --format json`), it's not an issue because the JSON output should not have any error output interleaved, and if it did, our new split output would be less confusing than the "native output" anyways.

Reply:
💭 What do you think of labeling the two outputs? 

```suggestion
        message: stderr_to_stdout ? stdout : "stderr: #{stderr}\n stdout: #{stdout}",
```

Reply:
I mostly copied this flag from `run_helper_subprocess` method:

https://github.com/dependabot/dependabot-core/blob/3373044e8ef289aa46789fce44a4bbb61319201d/common/lib/dependabot/shared_helpers.rb#L111-L115

Should we update that too?

Reply:
FWIW, I don't think this is a **need**. I suggested labeling because the question came up 😇 Please don't feel the need to change your approach based on my input!